grails-app/domain/org/pih/warehouse/order/OrderItem.groovy

```diff
@@ -320,16 +321,29 @@ class OrderItem implements Serializable, Comparable<OrderItem> {
                 : quantityShippedInStandardUom - quantityInvoicedInStandardUom
     }
 
+    /**
+     * Item can be on a regular invoice if it is not canceled or if it is canceled, but has prepayment
+     * */
+    Boolean canBeOnRegularInvoice() {
+        if (canceled) {
+            return hasPrepaymentInvoice
+        }
+
+        return true
```

Comment:
same here. Can be simplified to return !canceled || hasPrepaymentInvoice if you want